### ISSUE TITLE
Wrong result with expressions as the group keys

### ISSUE THREAD

**Issue by gfphoenix78** [greenplum-db/gpdb#10878]
### Greenplum version or build
6X

This issue is similar to https://github.com/greenplum-db/gpdb/issues/10720, but not the same.
### Expected behavior
gpadmin=# select j >= 0, not j >= 0 from multiagg_expr_group_tbl group by j >= 10, j >= 0;
 ?column? | ?column?
----------+----------
 f        | t
 t        | f
(2 rows)
### Actual behavior
gpadmin=# select j >= 0, not j >= 0 from multiagg_expr_group_tbl group by j >= 10, j >= 0;
 ?column? | ?column?
----------+----------
 f        | f
 t        | f
(2 rows)
### Step to reproduce the behavior
create table multiagg_expr_group_tbl (i int, j int) distributed by (i);
insert into multiagg_expr_group_tbl values(-1, -2), (-1, -1), (0, 1), (1, 2);
select j >= 0, not j >= 0 from multiagg_expr_group_tbl group by j >= 10, j >= 0;

```
gpadmin=# explain (verbose) select j >= 0, not j >= 0 from multiagg_expr_group_tbl group by j >= 10, j >= 0;
                                                                       QUERY PLAN
--------------------------------------------------------------------------------------------------------------------------------------------------------
 Gather Motion 3:1  (slice2; segments: 3)  (cost=2.24..2.29 rows=4 width=3)
   Output: ((multiagg_expr_group_tbl.j >= 0)), ((((multiagg_expr_group_tbl.j >= 0)) OPERATOR(pg_catalog.<) 0)), ((multiagg_expr_group_tbl.j >= 10))
   ->  HashAggregate  (cost=2.24..2.29 rows=2 width=3)
         Output: ((multiagg_expr_group_tbl.j >= 0)), (((multiagg_expr_group_tbl.j >= 0)) OPERATOR(pg_catalog.<) 0), ((multiagg_expr_group_tbl.j >= 10))
         Group Key: ((multiagg_expr_group_tbl.j >= 10)), ((multiagg_expr_group_tbl.j >= 0))
         ->  Redistribute Motion 3:3  (slice1; segments: 3)  (cost=2.08..2.18 rows=2 width=2)
               Output: ((multiagg_expr_group_tbl.j >= 10)), ((multiagg_expr_group_tbl.j >= 0))
               Hash Key: ((multiagg_expr_group_tbl.j >= 10)), ((multiagg_expr_group_tbl.j >= 0))
               ->  HashAggregate  (cost=2.08..2.10 rows=2 width=2)
                     Output: ((multiagg_expr_group_tbl.j >= 10)), ((multiagg_expr_group_tbl.j >= 0))
                     Group Key: (multiagg_expr_group_tbl.j >= 10), (multiagg_expr_group_tbl.j >= 0)
                     ->  Seq Scan on public.multiagg_expr_group_tbl  (cost=0.00..2.06 rows=2 width=4)
                           Output: multiagg_expr_group_tbl.j, (multiagg_expr_group_tbl.j >= 10), (multiagg_expr_group_tbl.j >= 0)
 Optimizer: Postgres query optimizer
(14 rows)
```

**Comment by gfphoenix78** [comment]
The result in the upstream is correct, it uses one-phase aggregate.
```
gpadmin=# select j >= 0, not j >= 0 from multiagg_expr_group_tbl group by j >= 10, j >= 0;
 ?column? | ?column?
----------+----------
 f        | t
 t        | f
(2 rows)

gpadmin=# explain verbose select j >= 0, not j >= 0 from multiagg_expr_group_tbl group by j >= 10, j >= 0;
                                       QUERY PLAN
----------------------------------------------------------------------------------------
 HashAggregate  (cost=55.20..55.27 rows=4 width=3)
   Output: ((j >= 0)), (j < 0), ((j >= 10))
   Group Key: (multiagg_expr_group_tbl.j >= 10), (multiagg_expr_group_tbl.j >= 0)
   ->  Seq Scan on public.multiagg_expr_group_tbl  (cost=0.00..43.90 rows=2260 width=6)
         Output: (j >= 0), (j >= 10), j
(5 rows)
```

**Comment by kainwen** [comment]
Fixed by a8b33fc13010d44312e956ac797858396ac8b9f9
